example/client/include/KVMessageGenerator.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <chrono>
+#include <map>
+#include <set>
+#include <vector>
+#include <cstdio>
+#include <cstring>
+#include <sys/param.h>
+#include <unistd.h>
+#include <yaml-cpp/yaml.h>
+#include <TimeUtils.hpp>
+
+#include "Logger.hpp"
+#include "kv_replica_msgs.cmf.hpp"
+#include "messages/ClientRequestMsg.hpp"
+#include "KVBCInterfaces.h"
+#include "common.hpp"
+#include "block_metadata.hpp"
+#include "boost/detail/endian.hpp"
+#include "assertUtils.hpp"
+#include "storage/db_types.h"
+#include "IMessageGenerator.hpp"
+
+namespace concord::kvbc {
+using namespace ::kvbc::messages;
+
+struct KVBlock {
+  uint64_t id = 0;
+  std::vector<std::pair<std::vector<uint8_t>, std::vector<uint8_t>>> items;
+};
+typedef std::map<std::pair<std::vector<uint8_t>, uint64_t>, std::vector<uint8_t>> KeyBlockIdToValueMap;
+
+class KVMessageGenerator : public IMessageGenerator {
+ public:
+  KVMessageGenerator(SampleMessage* sampleMsg) : sampleMsg_(sampleMsg) {}
+  ~KVMessageGenerator() = default;
+  void setMessageFileName(const std::string& fileName) { sampleMsgFileName_ = fileName; }
```

Comment:
can we set the filename at the time of creation using the ctor?

Reply:
removed. it wasn't used in KVMessageGenerator class.